core/src/main/java/org/mskcc/cbio/portal/servlet/PatientView.java

```diff
@@ -484,12 +485,39 @@ private void setClinicalInfo(HttpServletRequest request) throws DaoException {
         }
         
         // path report
-        String typeOfCancer = cancerStudy.getTypeOfCancerId();
-        if (patientId!=null && patientId.startsWith("TCGA-")) {
-            String pathReport = getTCGAPathReport(typeOfCancer, patientId);
-            if (pathReport!=null) {
-                request.setAttribute(PATH_REPORT_URL, pathReport);
+        if (GlobalProperties.useInternalPathReports()) {
+            // Expected url path 'studyId/patientId/sampleId.pdf' maps to expected filesystem path 'studyId.patientId.sampleId.pdf'
+            final String reportBaseURL = "pathology_report/" + cancerStudy.getCancerStudyStableId() 
+                + "/" + patientId + "/";
+            final String reportBasePath = GlobalProperties.getInternalPathReportRoot() + "/" 
+                + cancerStudy.getCancerStudyStableId() + "." + patientId + ".";
+            if (isPatientView) {
+                Map<String,String> sampleReportUrls = new HashMap<String,String>();
+                for (String sample : samples) {
+                    final String reportFileName = sample + ".pdf";
+                    if (new File(reportBasePath + reportFileName).exists()) {
+                        sampleReportUrls.put(sample, reportBaseURL + reportFileName);
+                    }
+                }
+                if (!sampleReportUrls.isEmpty()) {
+                    //ToDo: send across the map of all sample URLs and handle the map within the patient_view.jsp and the path_report.jsp
+                    final String firstSampleReport = sampleReportUrls.entrySet().iterator().next().getValue();
```

Comment:
Leave for now then, but in general anticipatory coding is a bad thing to do, because 99% of the time it will never get used (just take a look at all the commented code in Alveo that was left there "just in case it was going to be used in the future").